modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -271,6 +272,9 @@
     /** Last time received message from ring. */
     private volatile long lastRingMsgReceivedTime;
 
+    /** */
+    private volatile Boolean clusterSupportsTcpDiscoveryNodeSerializationOptimization = null;
```

Comment:
Server impl it's single threaded class. `volatile` doesn't required. You could use primitive `boolean` type. Initialize its in constructor and recalculate on nodeAdded/nodeLeft/nodeFailed events.

Reply:
fixed, also added nodeAdd event